DRA Node E2E: remove Serial label

bart0sh opened the pull request:
#### What type of PR is this?

/kind failing-test

#### What this PR does / why we need it:

Removed Serial label from the DRA tests to fix
cos-cgroupv1/v*-containerd-node-e2e-serial and other CI jobs that can be triggerd by this label.

#### Which issue(s) this PR fixes:

Fixes #118660

#### Special notes for your reviewer:

This is a follow-up PR for #118665.
#118665 partly fixed `kubelet-gce-e2e-arm64-ubuntu-serial` job

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

bart0sh committed:
DRA Node E2E: remove Serial label

Removed Serial label from the DRA tests to fix
cos-cgroupv1/v*-containerd-node-e2e-serial and other
CI jobs that can be triggerd by this label.

bart0sh commented:
/test pull-kubernetes-node-e2e-crio-dra

bart0sh commented:
/test pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial

bart0sh commented:
/test kubelet-gce-e2e-arm64-ubuntu-serial

pohly reviewed (commented):
I think this change is backwards: the `Serial` tag makes sense for these tests because they cannot be run in parallel, can they?

What should be fixed are the focus/skip parameters in the jobs which accidentally include these tests.

bart0sh commented:
@pohly The idea of this PR was to test if we can fix -serial jobs without changing all of them. btw, to test it properly we'll need to merge https://github.com/kubernetes/test-infra/pull/29811 to ensure that this PR doesn't break dra-node job.

bart0sh commented:
/hold until https://github.com/kubernetes/test-infra/pull/29811 is merged

bart0sh commented:
/test pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial
/test kubelet-gce-e2e-arm64-ubuntu-serial

bart0sh commented:
/test pull-kubernetes-cos-cgroupv1-containerd-node-e2e-serial

pohly commented:
> The idea of this PR was to test if we can fix -serial jobs without changing all of them

I still think that this is not the right solution for the problem. First each test should be labeled so that the labels match what the test is doing, then jobs should be defined to run tests based on these labels - not the other way around.

bart0sh commented:
> I still think that this is not the right solution for the problem. 

What's the alternative? Put `--skip "\[Feature:DynamicResourceAllocation\]` into **every** serial job?

pohly commented:
In E2E, normal serial jobs skip all alpha features. Failures in such jobs are more important because the features they test is supposed to be stable. This alone would prevent running the DRA tests, without a dedicated "skip DRA" in those jobs.

Then dedicated jobs deal with alpha features.

In addition, there are "feature tags" which presumably define what features the cluster must have to run a test. If configured correctly, that also would exclude DRA from jobs that aren't prepared for it.

bart0sh commented:
@pohly That makes sense, thanks. I'll close this PR and open a new one that adds NodeAlphaFeature:DynamicResourceAllocation to the DRA Node job.
does it make sense to you?

pohly commented:
And `NodeAlphaFeature` gets skipped by those jobs? Then that looks like the right solution.

pohly commented:
For now... we will still need some feature-based solution for the time when DRA is not alpha anymore.

bart0sh commented:
> And NodeAlphaFeature gets skipped by those jobs? Then that looks like the right solution.

Yes, failing containerd-serial jobs skip NodeAlphaFeature. 

Here is a list of the jobs that still may fail, but they're not failing now. Probably they're no triggered by dra code changes:
```
$ git grep 'focus=[^ ]*Serial' |grep -v 'skip=[^ ]*NodeAlphaFeature' |grep yaml | grep -v ginkgo.focus=Kubectl
config/jobs/kubernetes-sigs/cloud-provider-azure/cloud-provider-azure-config.yaml:        value: --ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Flaky\]|\[Feature:.+\]|regular.resource.usage.tracking.resource.tracking.for|validates.MaxPods.limit.number.of.pods.that.are.allowed.to.run|In-tree.Volumes|RuntimeClass.should.run.a.Pod.requesting.a.RuntimeClass.with.scheduling.with.taints|Multi-AZ.Clusters.should.spread.the.pods.of.a.replication.controller.across.zones|Multi-AZ.Clusters.should.spread.the.pods.of.a.service.across.zones|Should.test.that.pv.used.in.a.pod.that.is.deleted.while.the.kubelet.is.down.cleans.up.when.the.kubelet.returns|Should.test.that.pv.used.in.a.pod.that.is.force.deleted.while.the.kubelet.is.down.cleans.up.when.the.kubelet.returns|Should.test.that.pv.written.before.kubelet.restart.is.readable.after.restart|subPath.should.unmount.if.pod.is.force.deleted.while.kubelet.is.down|subPath.should.unmount.if.pod.is.gracefully.deleted.while.kubelet.is.down|Should.test.that.pv.written.before.kubelet.restart.is.readable.after.restart --report-dir=/logs/artifacts --disable-log-dump=true
config/jobs/kubernetes/generated/generated.yaml:      - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Driver:.gcepd\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
config/jobs/kubernetes/generated/generated.yaml:      - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Driver:.gcepd\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
config/jobs/kubernetes/generated/generated.yaml:      - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Driver:.gcepd\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
config/jobs/kubernetes/generated/generated.yaml:      - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Driver:.gcepd\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
config/jobs/kubernetes/sig-cli/sig-cli-config.yaml:      - --test_args=--ginkgo.focus=\[sig-cli\].*\[Serial\]|\[sig-cli\].*\[Disruptive\] --ginkgo.skip=\[Flaky\]|\[Feature:.+\] --minStartupPods=8
config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml:            - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Driver:.gcepd\]|\[Slow\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml:      - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Driver:.gcepd\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
config/jobs/kubernetes/sig-network/sig-network-misc.yaml:      - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Flaky\]|\[sig-storage\]|\[Feature:.+\] --minStartupPods=8
config/jobs/kubernetes/sig-network/sig-network-misc.yaml:      - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Flaky\]|\[sig-storage\]|\[Feature:.+\]|LoadBalancer --minStartupPods=8
config/jobs/kubernetes/sig-node/containerd.yaml:#      - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Driver:.gcepd\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
config/jobs/kubernetes/sig-release/release-branch-jobs/1.25.yaml:        - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Driver:.gcepd\]|\[Slow\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
config/jobs/kubernetes/sig-release/release-branch-jobs/1.26.yaml:        - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Driver:.gcepd\]|\[Slow\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
config/jobs/kubernetes/sig-release/release-branch-jobs/1.27.yaml:        - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Driver:.gcepd\]|\[Slow\]|\[Flaky\]|\[Feature:.+\] --minStartupPods=8
config/jobs/kubernetes/sig-storage/sig-storage-gce-config.yaml:        - --test_args=--ginkgo.focus=CSI.*(\[Serial\]|\[Disruptive\]) --ginkgo.skip=\[Flaky\]|\[Feature:.+\]|\[Slow\] --minStartupPods=8
config/jobs/kubernetes/sig-storage/sig-storage-gce-config.yaml:      - --test_args=--ginkgo.focus=\[Driver:.iscsi\].*(\[Serial\]|\[Disruptive\]) --ginkgo.skip=\[Flaky\] --minStartupPods=8
releng/test_config.yaml:    - --test_args=--ginkgo.focus=\[Serial\]|\[Disruptive\] --ginkgo.skip=\[Driver:.gcepd\]|\[Flaky\]|\[Feature:.+\]
```


> For now... we will still need some feature-based solution for the time when DRA is not alpha anymore.

Yes, I'm going to backport CDI support to Containerd 1.6 and use it for test-infra jobs when CDI support is released. This would help to prepare for DRA beta, I think.

bart0sh commented:
@pohly Note, that `pull-kubernetes-node-e2e-crio-dra` and `pull-kubernetes-cos-cgroupv2-containerd-node-e2e-serial` succeeded, so this solution (removing Serial label) is not that bad.

bobbypage commented:
this sounds good to me as solution for now. Another option would be to have the dra_test explicitly skip the test if the environment is not setup as intended.  

/lgtm

bart0sh commented:
@bobbypage let's do as @pohly suggested above, here is an alternative PR: https://github.com/kubernetes/kubernetes/pull/118721

bart0sh commented:
closing in favour of https://github.com/kubernetes/kubernetes/pull/118721
